hw/formal/tools/jaspergold/parse-fpv-report.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+"""
+Parses JasperGold FPV logfile and coverage and dump filtered messages in hjson format.
+"""
+
+import argparse
+import re
+from pathlib import Path
+from collections import OrderedDict
+
+import hjson
+
+
+def extract_messages(full_file, patterns, results):
+    """
+    This extracts messages from the sting buffer full_file.
+    The argument patterns needs to be a list of tuples with
+    (<key>, <pattern_to_match_for>).
+    """
+    for severity, pattern in patterns:
+        results[severity] = re.findall(pattern, full_file, flags=re.MULTILINE)
+
+    return results
+
+
+def extract_messages_count(full_file, patterns, results):
```

Comment:
Similar comments to `extract_messages`: docstring formatting and possible unnecessary `results` argument.